Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -221,8 +221,39 @@ - (void)_registerForNotifications
 #if HAVE(UIKIT_RESIZABLE_WINDOWS)
     [center addObserver:self selector:@selector(_enhancedWindowingToggled:) name:_UIWindowSceneEnhancedWindowingModeChanged object:nil];
 #endif
+#if PLATFORM(VISION)
+    [center addObserver:self selector:@selector(_webViewSceneWillEnterForeground:) name:UISceneWillEnterForegroundNotification object:nil];
+    [center addObserver:self selector:@selector(_webViewSceneDidActivate:) name:UISceneDidActivateNotification object:nil];
+    [center addObserver:self selector:@selector(_webViewSceneWillEnterBackground:) name:UISceneDidEnterBackgroundNotification object:nil];
+    [center addObserver:self.window selector:@selector(_webViewWindowDidBecomeKey:) name:UIWindowDidBecomeKeyNotification object:nil];
```

Comment:
`addObserver:self.window` seems incorrect – shouldn't it be `addObserver:self`?

Also, should we use a non-nil object (either the scene or window), depending on the notification, so that we're not listening for notifications from objects we don't care about?

Reply:
Oh yes it should be self and good idea, I will use a non-nil object

Reply:
Actually I'm going to remove the notifications. They don't occur at the times I need and tracking the state and setting it later seems unnecessary.